src/problem.jl

```diff
@@ -0,0 +1,147 @@
+abstract type AbstractProblem end
+
+"""
+    compute(problem, grid::Union{Grid,GridRSP})
+
+Compute problem `o` on a grid.
+"""
+function compute end
+
+"""
+    assess(p::AbstractProblem, g)
+
+Assess the memory and compute requirements of problem
+`p` on grid `g`. This can be used to indicate memory
+and time reequiremtents on a cluster
+"""
+function assess end
+
+"""
+    SolverMode
+
+Abstract supertype for ConScape solver modes.
+"""
+abstract type SolverMode end
+"""
+   MaterializedSolve()
+
+Compute all operations on a fully materialised Z matrix.
+
+This is inneficient for CPUS, but may be best for GPUs
+using CuSSP.jl ?
+"""
+struct MatrixSolve <: SolverMode end
+"""
+   ColumnSolve()
+
+Compute all operations column by column,
+after precomputing lu decompositions.
+
+TODO we can put LinearSolve.jl solver objects in the struct
+"""
+struct VectorSolve <: SolverMode end
+
+"""
+    Problem(ops...; mode, θ)
+
+Combine multiple compute operations into a single object, 
+to be run in the same job.
+"""
+@kwdef struct Problem{O,M<:SolverMode,T} <: AbstractProblem
+    ops::O
+    mode::M=VectorSolve()
+    θ::T=1.0
+end
+Problem(ops::AbstractOperation...; kw...) = Problem(ops; kw...)
+Problem(ops::Tuple; kw...) = Problem(; ops, kw...)
+Problem(ops::NamedTuple; kw...) = Problem(; ops, kw...)
+Problem(p::AbstractProblem; mode=p.mode, θ=nothing) = Problem(o.ops, mode, θ)
+
+compute(p::Problem, rast::RasterStack) = compute(p, Grid(p, rast))
+compute(p::Problem, g::Grid) = compute(p.mode, p, g::Grid)
+
+# Use an iterative solver so the grid is not materialised
+function compute(m::VectorSolve, p::AbstractProblem, g::Grid)
+    # Compute Z column by column
+    _, targetnodes = _targetidx_and_nodes(g)
+    Pref = _Pref(g.affinities)
+    W = _W(Pref, p.θ, g.costmatrix)
+    # Sparse lhs
+    A = I - W
+    # Sparse diagonal rhs matrix
+    B = sparse(targetnodes,
+        1:length(targetnodes),
+        1.0,
+        size(g.costmatrix, 1),
+        length(targetnodes),
+    )
+    # Dense rhs column
+    b = zeros(eltype(A), size(B, 1))
+    @show size(A) size(B) size(b)
+
+    # Define and initialise the linear problem
+    linprob = LinearProblem(A, b)
```

Comment:
See my comment above, this should appear in the construction of the Problem. We should additionally allow the user to provide a solver, with possibly some warnings or advices on suitable solvers

Reply:
For sure, this is just hacked in for now to get it working while I learn LinearSolve.

Reply:
Something that should also be addressed, and that I do not fully understand: in many of the methods in `randomizedshortestpath.jl`, there is a second linear solve with the use of landmarks 

https://github.com/ConScape/ConScape.jl/blob/c18ca9b601f23182f4f8e06351100a441c5298bc/src/randomizedshortestpath.jl#L194

This may have consequences on how we want to construct a `Problem`.

Reply:
Yeah I saw that too. These methods will need to be handled but I don't totally understand them yet either.